osgp/platform/osgp-core/src/test/java/org/opensmartgridplatform/core/application/tasks/ScheduledTaskExecutorServiceTest.java

```diff
@@ -102,4 +104,65 @@ public void testRunFunctionalException()
     // check if task is deleted
     verify(this.scheduledTaskRepository).delete(scheduledTask);
   }
+
+  @Test
+  void testRetryStrandedPendingTask() {
+
+    final List<ScheduledTask> scheduledTasks = new ArrayList<>();
+    final Timestamp scheduledTime = new Timestamp(Calendar.getInstance().getTime().getTime());
+    final ScheduledTask expiredScheduledTask =
+        new ScheduledTask(
+            this.createExpiredMessageMetadata(), DOMAIN, DOMAIN, DATA_OBJECT, scheduledTime);
+    expiredScheduledTask.setPending();
+    scheduledTasks.add(expiredScheduledTask);
+    final ScheduledTask retryableScheduledTask =
+        new ScheduledTask(
+            this.createMessageMetadata(), DOMAIN, DOMAIN, DATA_OBJECT, SCHEDULED_TIME);
+    retryableScheduledTask.setPending();
+    scheduledTasks.add(retryableScheduledTask);
+    when(this.scheduledTaskExecutorJobConfig.scheduledTaskPendingDurationMaxSeconds())
+        .thenReturn(0L);
+    when(this.scheduledTaskExecutorJobConfig.scheduledTaskPageSize()).thenReturn(30);
+    when(this.scheduledTaskRepository.findByStatusAndScheduledTimeLessThan(
+            eq(ScheduledTaskStatusType.PENDING), any(Timestamp.class), any(Pageable.class)))
+        .thenReturn(scheduledTasks);
+    when(this.scheduledTaskRepository.findByStatusAndScheduledTimeLessThan(
+            eq(ScheduledTaskStatusType.NEW), any(Timestamp.class), any(Pageable.class)))
+        .thenReturn(new ArrayList<ScheduledTask>());
+    when(this.scheduledTaskRepository.findByStatusAndScheduledTimeLessThan(
+            eq(ScheduledTaskStatusType.RETRY), any(Timestamp.class), any(Pageable.class)))
+        .thenReturn(new ArrayList<ScheduledTask>());
```

Comment:
Have you considered refactoring this rather long test setup into multiple smaller methods?

Reply:
refactored this test method